src/Controllers/Common/ContentController.cs

```diff
@@ -523,7 +523,8 @@ public IActionResult SetAvatar(Viking viking, [FromForm] string contentXML) {
         raisedPetData.PetTypeID = petTypeID;
         raisedPetData.RaisedPetID = 0; // Initially make zero, so the db auto-fills
         raisedPetData.EntityID = Guid.Parse(dragonId);
-        raisedPetData.Name = string.Concat("Dragon-", dragonId.AsSpan(0, 8)); // Start off with a random name
+        if ((gameVersion & ClientVersion.WoJS) == 0)
```

Comment:
you should check also for `Max_OldJS` ... for example `gameVersion > ClientVersion.Max_OldJS || (gameVersion & ClientVersion.WoJS) == 0` or just `gameVersion >= ClientVersion.SS`